terraform_files/baremetal/main.tf

```diff
@@ -158,50 +158,55 @@ module "workers" {
 # Terraform doesn't have ability for conditional blocks (if cond { block }) so we're using
 # the count directive to include/exclude elements
 
+locals {
+  base_cluster_domain = var.base_cluster_domain == "" ? "${var.cluster_name}.${var.cluster_domain}" : var.base_cluster_domain
+}
+
 data "libvirt_network_dns_host_template" "api" {
   # API VIP is always present. A value is set by the installation flow that updates
   # either the single node IP or API VIP, depending on the scenario
   count    = 1
   ip       = var.bootstrap_in_place ? var.single_node_ip : var.api_vip
-  hostname = "api.${var.cluster_name}.${var.cluster_domain}"
+  hostname = "api.${local.base_cluster_domain}"
 }
 
 data "libvirt_network_dns_host_template" "api-int" {
   count    = 1
   ip       = var.bootstrap_in_place ? var.single_node_ip : var.api_vip
-  hostname = "api-int.${var.cluster_name}.${var.cluster_domain}"
+  hostname = "api-int.${local.base_cluster_domain}"
 }
 
-# TODO: leave only the wildcard cname entry defined and remove the other specific DNS assignments
+# TODO: leave only the wildcard address entry defined and remove the other specific DNS assignments
 # Read more at: https://bugzilla.redhat.com/show_bug.cgi?id=1532856
-data "libvirt_network_dnsmasq_options_template" "wildcard-ingress-cname" {
-  count        = var.master_count == 1 ? 1 : 0
-  option_name  = "cname"
-  option_value = "*.apps.${var.cluster_name}.${var.cluster_domain},${var.ingress_vip}"
+data "libvirt_network_dnsmasq_options_template" "wildcard-apps-ingress" {
+  # Enable "apps" wildcard in case of SNO and when we try to add day2 worker to SNO
+  count        = var.ingress_vip == var.api_vip ? 1 : 0
```

Comment:
I take the assumption that the condition `var.ingress_vip == var.api_vip` is a SNO cluster, is that always true?

Reply:
in our infra it's probably always true
although I guess there are other ways to assert SNO cluster (num_masters == 1)

Reply:
yeah, otherwise I need to pass an extra flag. What is this `single_node_ip` variable above, could it be reused here?

Reply:
yes I think so

Reply:
if anything I would prefer to move to more general values (api/ingress rather than having a separate var for SNO IP) but for now it wouldn't matter match. Only thing important here is not to add more flags as we already have too much IMHO :)

Reply:
here I need to know if the day1 is an SNO in order to set the DNS or not in the day2 network, at terraform level I have almost no information about day1 cluster except api and ingress vips.

Reply:
(I agree that we should not add more flags :) )

Reply:
ack. I missed this is in another context 
then the api==ingress is probably the best thing you can do ATM